### DISCUSSION TITLE
bizonylatok, xml formázás, tartalom, stb...

### DISCUSSION THREAD

**Discussion by cogentax** [nav-gov-hu/M2M#31] [Általános]
Sziasztok,

Egy általános kérdésem lenne. Eddig úgy működött, hogy a NAV kiadta/frissítette a nyomtatványokat, azokat megette az ÁNYK.  
Az ÁNYK-n keresztül, vagy segítségével elkészült egy ellenőrzött xml( vagy nyomtatvány)... majd beküldésre került az ÁNYK keresztül, vagy parancssorosan, stb...

A kérdésem az lenne, hogy az m2m esetében egyenként kell előállítani minden beküldendő bizonylathoz tartozó xml-t?  + érdemes, már elővalidálni az xml tartalmát, hogy ne az m2m api validációján akadjon el, és dobja vissza. Ez nem minden esetben egyszerű feladat, beleértve helyes közterültek nevét, stb... 

Lesz NAV által kiadott xml formátum minden bizonylathoz?

Meddig lesz még ÁNKY?

Köszi!

**Comment by MLaccc** [comment]
Sziasztok!

Nekem is ez lenne a kérdésem. Kiegészítve azzal, hogy az ANYK-ban rendszeresen jöttek új nyomtatványverziók, amelyek újabb validációt jelentettek, vagy pl. esetleg új mezőket., lapokat. Ezt állandóan le kellene programozni?  Eddig minden nyomtatványnak volt "nyomtatvány képe". Az M2M rendszerben akkor ez is megszűnik? Nem látom azt a kapcsolatot amit eddig a nyomtatvány enyk file tartott a felhasználó és a NAV között a nyomtatvány tartalmi követelményeivel szemben.

A választ előre is köszönöm!

**Comment by Szildiko** [comment]
Sziasztok!

Engem a tesztelői oldala érdekelne, én nem programozó vagyok, hanem akinek majd ezt tesztelni kéne. Az OSA rendszer kialakítása átlátható volt számomra, de ezzel kapcsolatban nem látom ugyanazt a szisztémát. Hol érhető el a teszt rendszer, teszt rendszer felülete? Ha beküldök egy bevallást a teszt rendszerbe, hol tudom megnézni? Felhasználó kezelés itt is elérhető-e, mint az OSA-nál, ha regisztrálunk, akkor lehet-e több felhasználót hozzáadni a teszt rendszerhez? Ennek a teszt rendszernek nem ideiglenesnek kéne lennie, mert minden évben új nyomtatványok jönnek, amit le is kell majd tesztelni.

Előre is köszönöm a válaszokat!

**Reply by nanassy72** [reply]
A tesztrendszernek nincs felülete, csak API-n keresztül érhető el  a m2m-dev.nav.gov.hu végponton.

A tesztrendszerbe az ÜPO felületén az M2M regisztráció/Fejlesztői regisztráció/Tesztkörnyezet regisztráció alatt lehet regisztrálni. Ekkor a felületen jelennek meg a tesztkliens és tesztfelhasználó adatai, nem tárhelyüzenetben érkeznek, mint az éles regisztrációnál. 

Egy ÜPO felhasználó csak egy M2M tesztkliens-tesztfelhasználó párost tud generálni. Ez egy fejlesztők számára szánt tesztkörnyezet, ami az interfészműveletek alapvető kipróbálására szolgál, nem összetettebb tesztelésre.

A tesztrendszerben beküldött bevallások és nyugtáik nem tekinthetők meg. A beküldés válaszüzenete mutatja a művelet sikerességet.

A tesztrendszer nem ideiglenes, tartósan fennmarad.

**Comment by Almito420** [comment]
Sziasztok!

Szeretnék én is csatlakozni az egyeztetéshez. Átnéztük az API leírást és nem találtuk meg azokat a végpontokat ahonnan lehetséges lekérdezni a nyomtatvány sablonok XML fájljait, verzióit, lenyomatait, vagy azt, hogy az adott sablon még használható-e beküldésre.

Nagyon köszi a válaszokat!

**Reply by nanassy72** [reply]
Az M2M API jelenleg nem támogatja az aktuális nyomtatványinformáció lekérdezését. Szükség lesz ilyen szolgáltatásra, de ez még átgondolás alatt áll.

**Comment by nanassy72** [comment]
Sziasztok!

Itt egyszerre válaszolok az első két bejegyzés egymást részben átfedő kérdéseire.

**Az m2m esetében egyenként kell előállítani minden beküldendő bizonylathoz tartozó xml-t?**

Igen. Minden bizonylatot xml formában kell előállítani és beküldeni.

**Az ANYK-ban rendszeresen jöttek új nyomtatványverziók, amelyek újabb validációt jelentettek, vagy pl. esetleg új mezőket., lapokat. Ezt állandóan le kellene programozni?** 
Igen, a változásokat át kell vezetni az xml-t előállító kódban.

**Érdemes már elővalidálni az xml tartalmát, hogy ne az m2m api validációján akadjon el, és dobja vissza. Ez nem minden esetben 
egyszerű feladat, beleértve helyes közterültek nevét, stb...**

A fejlesztő cégek egy része eddig az ÁNYK-t alkalmazta (általában parancssori módban meghívva) „újraszámításra” és „validálásra”, mások saját maguk fejlesztették le a bizonylatot. „Elővalidálásról” értelmezésünk szerint csak az ÁNYK-s működési mód esetén van értelme beszélni. Mivel az M2M és az ÁNYK validációja lényegileg megegyeznek, tehát az ÁNYK parancssori mód továbbra is jól használható a bizonylatok „elővalidálására” – de ez természetesen nem kötelező. Azt eldönteni, hogy adott megoldás esetében az ÁNYK-s „előellenőrzés” kialakításának van-e akkora hozzáadott értéke, ami a szükséges befektetéssel arányban áll, tehát érdemes-e ezt alkalmazni, nem áll módunkban: ezt csak a rendszer fejlesztője tudja megítélni. 
 
**Lesz NAV által kiadott xml formátum minden bizonylathoz?**
 
Ha a kérdés a bizonylatok XML szerkezetét leíró XSD-re vonatkozik, akkor: egyetlen XSD séma van, ami az összes bizonylatnál alkalmazott általános XML szerkezetet írja le, és nem tervezünk bizonylattípusonkénti XSD-ket kiadni. 
 
Ha a kérdés a nyomtatványsablonok XML formátumban való publikálására (*.tem.enyk állományok) vonatkozik: A nav.gov.hu portál nyomtatványkeresőjéből az ÁNYK-ban „letiltott” nyomtatványokra vonatkozó információk eltávolítása téves intézkedés volt, és visszavonásra kerül. Így a *.tem.enyk állományok ezen nyomtatványok vonatkozásában is elérhetők maradnak. 
 
**Meddig lesz még ÁNYK?**
 
Az ÁNYK kivezetésével kapcsolatban egyelőre nem került konkrét időpont kijelölésre. Az ÁNYK kivezetése előtt a piacot időben értesíteni fogjuk úgy, hogy elegendő idő álljon rendelkezésre majd a felkészülésre. A NAV törekszik továbbá arra, hogy az ÁNYK funkcionalitásából azokat a funkciókat, melyek fenntartása a jövőben is indokolt lehet, az ÁNYK kivezetését megelőzően más megoldással kiváltsa.  
 
**Eddig minden nyomtatványnak volt "nyomtatvány képe". Az M2M rendszerben akkor ez is megszűnik?**
 
A nyomtatványok képét az ÁNYK-ban eddig meg lehetett jeleníteni, például az ÁNYK parancssori indításával. Ez a lehetőség továbbra is fennáll. 
 
**Nem látom azt a kapcsolatot, amit eddig a nyomtatvány enyk file tartott a felhasználó és a NAV között a nyomtatvány tartalmi követelményeivel szemben.**
 
Az M2M-en végzett bizonylat ellenőrzés is a *.tem.enyk fájlokban leírt szabályokon alapul, így ezek továbbra is használhatók a fejlesztői oldalon „specifikációként”. (Ld. továbbá a "Lesz NAV által kiadott xml formátum minden bizonylathoz?" kérdésre adott választ is)
